### DISCUSSION TITLE
Deploying via rest api

### DISCUSSION THREAD

**Discussion by SamShanks1** [vercel/vercel#8203] [Help] [answered]
Hi, 

Im attempting to deploy a site via the rest api

I am getting this error 

` 
    error: {
      code: 'missing_project_settings',
      message: 'The `projectSettings` object is required for new projects, but is missing in the deployment payload',
      framework: [Object],
      projectSettings: [Object]
    }
  }
  `

I cant seem to find any documentation on what should be in the projectSettings object

**Comment by SamShanks1** [comment, marked as answer]
I managed to work it out by looking through the source code and finding the projectSettings type.

For anyone else looking for this

the type can be found here: https://github.com/vercel/vercel/blob/fd9142b6f301625b14730e70b98e5bd3eb15bf0c/packages/build-utils/src/types.ts#L324

**Comment by amyegan** [comment]
Hi @SamShanks1. Thanks for reaching out! The `projectSettings` object is required for the first deployment of a new project and optional on future deployments. It should be part of the request body and look something like this:
```
"projectSettings": {
    "buildCommand": null,
    "devCommand": null,
    "framework": "nextjs",
    "commandForIgnoringBuildStep": "",
    "installCommand": null,
    "outputDirectory": null
},
```

https://vercel.com/docs/rest-api#endpoints/deployments/create-a-new-deployment

**Comment by defiveninth** [comment]
hello
